token/core/zkatdlog/nogh/tokens.go

```diff
@@ -128,3 +140,79 @@ func supportedTokenFormat(pp *crypto.PublicParams) (token.Format, error) {
 	}
 	return token.Format(hasher.HexDigest()), nil
 }
+
+func (s *TokensService) ProcessTokenConversionRequest(utp *driver.TokenConversionRequest) ([]token.Type, []uint64, error) {
+	if utp == nil {
+		return nil, nil, errors.New("nil token conversion request")
+	}
+
+	ok, err := s.CheckConversionProof(utp.Proof, utp.Proof, utp.Tokens)
```

Comment:
`utp.Challenge`

Reply:
I'll introduce types to avoid this issue